### ISSUE TITLE
Declaring dependencies

### ISSUE THREAD

**Issue by d-stahl-ericsson** [Ericsson/eiffel#10]
One use case that is not currently addressed is declaration of dependencies. I have seen this done using Eiffel in rather home grown ways... I feel as though I haven't quite thought this through, but I'm throwing it out there. The use case is essentially this:

Not only do I need to declare the composition of my artifact, but I also want to declare its dependencies so that I can check whether a higher level composition, containing the artifact, is actually legal (i.e. containing the dependencies declared by the artifact). 

The main issue I see is how to deal with version ranges here - you don't want to point to a range of versions, sort of like Maven dependencies. Or you could go convention-heavy and essentially expect semantic versioning. Or point to a specific version (i.e. event id) and expect the reader (with an understanding of one's particular versioning scheme) to make sense of the version range that should be acceptable. And speaking of Maven, we're on stepping into territory where I'm afraid we'll end up re-inventing the wheel.

**Comment by d-stahl-ericsson** [comment]
This will need to be resolved in no later than drop2, as it included CompositionDefined.

**Comment by d-stahl-ericsson** [comment]
I'm currently seeking feedback from a number of sources and am planning a workshop regarding this. Current thinking is that this is not so much a CompositionDefined issue, but an ArtifactCreated issue: it's not the composition's business to tell its elements what they need. The composition should, however, be useful in determining legality based on what the constituent artifacts have declared.

In summary, this boils down to a likely addition to ArtifactCreated at a later date, possibly as part of drop3.
